### DISCUSSION TITLE
Feature request (Vertical scrolling)

### DISCUSSION THREAD

**Discussion by SomeRandoLameo** [YaLTeR/niri#757] [Ideas]
Hello developers!

I saw this project and thought to me, man. I really need this, however there is one thing that stood in the way between me installing it.
Does Niri support horizontal scrolling? 
If not, I'd like to contribute (;

**Comment by YaLTeR** [comment]
Do you mean mirroring the layout into portrait mode so that new windows appear below?

**Comment by SomeRandoLameo** [comment]
Exactly, so that you can basically scroll up or down to move the windows or down instead or right or left

**Reply by YaLTeR** [reply]
So this is not currently implemented, and while possible, it would likely be quite an annoying refactor.

**Reply by SomeRandoLameo** [reply]
I'd be willing to change that in a commit. This might take some time since I am not so familiar with this project.

**Reply by YaLTeR** [reply]
I'm afraid it will take... quite a bit of time. I'm not even sure myself what the best way to refactor the code would be.

**Comment by Linguiniotta** [comment]
+1. 

Thank you for this wonderful project. Also hoping for this feature as I am looking to something like [hyprscroller](https://github.com/dawsers/hyprscroller), which allows both modes to coexist at the same time (scrolling vertically when window is admitted to the column, while still allowing scrolling horizontally).

**Comment by chilledfrogs** [comment]
I see this addition to the project (in combination with touchscreen gesture support) as an excellent possibility for mobile devices such as the PinePhone Pro, and therefore was thinking at taking a stab at this myself, but I fear my Rust knowledge isn't that good yet, but in any case it would make a *lot* of sense for portrait monitors to by default have a flipped layout (i.e. up/down for windows in a workspace, and left/right for switching workspaces), togglable dynamically of course if need be

**Comment by balsoft** [comment]
I took a stab at implementing this, and sadly it looks like it's way more difficult than one can imagine. A lot of functions in `src/layout/scrolling.rs` (and probably elsewhere) rely on the width & X position of everything, so changing it to height & Y position would require non-trivial refactoring.

However, for my use-case (two monitors, one landscape and one portrait) it would already be a great improvement if I could set `default-column-width` per-output (e.g. `proportion 1.0` on the portrait one, `proportion 0.5` on a landscape one). I feel it'd be somewhat easier to implement, I'll take a look at it next.

**Reply by balsoft** [reply]
Here's a draft for this idea: https://github.com/YaLTeR/niri/pull/973

**Comment by tv42** [comment]
I have a portrait monitor on the right side of my primary ultra-wide monitor (it's quite far to the right so portrait keeps it closer for ergonomics, and I'd run out of desk space too). Niri is great on the ultrawide, but vertical columns on a portrait monitor are a) hilarious and b) unusable. I end up making every column full width on the portrait monitor, for now, but I'd love to have niri manage the space more efficiently -- basically, rotate the "scroll of columns".

**Comment by YaLTeR** [comment]
Well let's make an issue for the +1s I guess: https://github.com/YaLTeR/niri/issues/1071

I'm afraid the code changes required for this didn't get any easier though since my last comment
